.github/workflows/components.yaml

```diff
@@ -43,3 +43,61 @@ jobs:
 
       - name: Build
         run: yarn build
+
+  build-and-deploy-storybook:
+    name: Deploy components to storybook
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build storybook
+        run: yarn build-storybook
+
+      # A pull_request event is a PR; deploy to preview testnet bucket.
+      - name: Deploy PR storybook preview to GCP
+        if: github.event_name == 'pull_request'
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.STORYBOOK_UPLOADER_SERVICE_KEY_JSON_BASE64 }}
+          project: ${{ secrets.STORYBOOK_GOOGLE_PROJECT_ID }}
+          bucket-name: preview.storybook.threshold.network
+          bucket-path: ${{ github.head_ref }}
+          build-folder: storybook-static
+
+      - name: Post preview URL to PR
+        if: github.event_name == 'pull_request'
+        uses: actions/github-script@v5
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Preview uploaded to https://preview.storybook.threshold.network/${{ github.head_ref }}/index.html.'
+            })
+
+      # A push event is triggered on main branch merge; deploy to testnet
+      # bucket. Also triggered by manual dispatch from `main` branch.
+      - name: Deploy storybook to GCP bucket
+        if: |
+          github.event_name == 'push'
+          || (github.event_name == 'workflow_dispatch'
+          && github.ref == 'refs/heads/main')
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.STORYBOOK_UPLOADER_SERVICE_KEY_JSON_BASE64 }}
+          project: ${{ secrets.STORYBOOK_GOOGLE_PROJECT_ID }}
```

Comment:
Or maybe we can use here the same secrets which we use when deploying T dashboard? I.e.
```
gcpServiceKey: ${{ secrets.KEEP_TEST_CI_UPLOAD_DAPP_JSON_KEY_BASE64 }}
gcpProject: ${{ secrets.KEEP_TEST_GOOGLE_PROJECT_ID }}
```